### ISSUE TITLE
Rule Change: (support message for path & patterns format in no-restricted-imports)

### ISSUE THREAD

**Issue by juanmait** [eslint/eslint#15651]
### What rule do you want to change?

no-restricted-imports

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

Other

### Example code

```js
{
  'no-restricted-imports': [
    'error',
    {
      paths: ['lodash', 'lodash-es'],
      patterns: ['lodash/*', '!lodash-es/*'],

      // not supported
      message: 'Please use only "lodash-es" submodules. E.g: "lodash-es/isPlainObject"'
    }
  ]
}
```

### What does the rule currently do for this code?

Configuration for rule "no-restricted-imports" is invalid. Value:

```json
{
  "paths": ["lodash", "lodash-es"],
  "patterns": ["lodash/*", "!lodash-es/*"],
  "message": "Please use only \"lodash-es\" submodules. E.g: \"lodash-es/isPlainObject\""
}
```

- Value should be string.
- Value should NOT have additional properties.
- Value should match some schema in anyOf.
- Value should NOT have additional properties.

### What will the rule do after it's changed?

Should provide this message:

```
import is restricted from being used by a pattern. Please use only "lodash-es" submodules. E.g: "lodash-es/isPlainObject"
```

For the following imports:

```js
import foo from 'lodash';
import bax from 'lodash/fill';
import bar from 'lodash-es';
```

But not for this one:

```js
import isPlainObject from 'lodash-es/isPlainObject'
```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

I tried to accomplish the same result using other patterns like path name & groups but none of that works. For example this one does not work either:

```json
{
  "no-restricted-imports": [
    "error",
    {
      "patterns": [
        {
          "group": ["lodash", "lodash-es", "!lodash-es/*"],
          "message": "Please use only \"lodash-es\" submodules. E.g: \"lodash-es/isPlainObject\""
        }
      ]
    }
  ]
}
```
It still reports errors with this import. Seems like the pattern `'!lodash-es/*'` does not work.

```js
import isPlainObject from 'lodash-es/isPlainObject'
```

**Comment by juanmait** [comment]
@mdjermanovic That works! Didn't know about the `.gitignore` algorithm. Thanks for the help!
